crates/api/src/runtime.rs

```diff
@@ -535,13 +536,18 @@ impl Engine {
 /// ocnfiguration (see [`Config`] for more information).
 #[derive(Clone)]
 pub struct Store {
-    // FIXME(#777) should be `Arc` and this type should be thread-safe
     inner: Rc<StoreInner>,
 }
 
-struct StoreInner {
+pub(crate) struct StoreInner {
     engine: Engine,
     compiler: RefCell<Compiler>,
+    instances: RefCell<Vec<InstanceHandle>>,
+}
+
+pub struct StoreInstanceHandle {
```

Comment:
Is this meant to be exported from the crate? It doesn't seem to be in lib.rs. If not, can this be `pub(crate)`?